rest-server/src/templates/yarnContainerScript.mustache

```diff
@@ -73,6 +73,92 @@ docker login --username ${cred[1]} --password ${cred[2]} ${cred[0]} \
   || { echo 'Authorized failed' ; exit 1; }
 {{/ jobData.authFile }}
 
+
+# Get container host list from framework launcher webservice
+function get_host_list()
+{
+  delay_time=1
+  while true; do
+    taskrole_statuses=$(curl -i -L -H "Accept: application/json" {{{ frameworkInfoWebhdfsUri }}} | jq -r ".aggregatedFrameworkStatus.aggregatedTaskRoleStatuses")
+    jq -re ".[].taskStatuses.taskStatusArray[].containerIp" <<< $(echo $taskrole_statuses) > /dev/null
+    if [ $? -eq 0 ]; then
```

Comment:
If cannot find any containerIp, we should also wait and looping.

Reply:
The taskrole_statuses may changed and only changed from EMPTY(0)  to the taskrole_statuses which contains the requested tasknumber. So, no intermediate state between 0 and requested tasknumber. However, you should at least check the 0 case.

---
In reply to: [192622034](https://github.com/Microsoft/pai/pull/467#discussion_r192622034) [](ancestors = 192622034)